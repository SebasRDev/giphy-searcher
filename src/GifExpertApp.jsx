import { useState } from "react"
import AddCategory from "./components/AddCategory"
import GifGrid from "./components/GifGrid"

const GifExpertApp = () => {

  const [categories, setCategories] = useState([])

  return(
    <div>
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={setCategories}/>
      <hr/>
      <ol>
        {
          categories.map(x => 
            <GifGrid 
              category={x}
              key={x}  
            />
          )
        }
      </ol>
    </div>
  )
}

export default GifExpertApp