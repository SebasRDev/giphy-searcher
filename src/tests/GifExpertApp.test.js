import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('test over GifExpertApp Component', () => {

  test('should match with snapshot', () => { 
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  })

  test('should show a categories list', () => { 
    const categories = ['one punch', 'dragon ball']
    const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
    
    expect(wrapper).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length).toBe(categories.length)
  })
})