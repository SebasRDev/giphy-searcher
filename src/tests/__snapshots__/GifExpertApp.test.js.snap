// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test over GifExpertApp Component should match with snapshot 1`] = `
<div>
  <h2>
    Buscador de Gifs
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;

exports[`test over GifExpertApp Component should show a categories list 1`] = `
<div>
  <h2>
    Buscador de Gifs
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="one punch"
      key="one punch"
    />
    <GifGrid
      category="dragon ball"
      key="dragon ball"
    />
  </ol>
</div>
`;
